packages/spectral/src/serverTypes/convertIntegration.ts

```diff
@@ -184,7 +184,16 @@ const convertComponentReference = <TValue>({
 
   const inputs = Object.entries(values ?? {}).reduce((result, [key, value]) => {
     if ("value" in value) {
-      return { ...result, [key]: { type: "value", value: value.value } };
+      const type = value.value instanceof Object ? "complex" : "value";
+      const valueExpr =
+        value.value instanceof Object && !Array.isArray(value.value)
+          ? Object.entries(value.value).map<ServerInput>(([key, value]) => ({
```

Comment:
`key` and `value` are already in the current scope due to the top-level `reduce`, are we comfortable with shadowing here?